metric-publishers/emf-metric-publisher/src/main/java/software/amazon/awssdk/metrics/publishers/emf/MetricEmfConverter.java

```diff
@@ -0,0 +1,368 @@
+    }
+
+    /**
+     * Processes and normalizes metric values for EMF formatting.
+     * The method handles various input types and normalizes them according to EMF requirements:
+     * - `null` : 0.0
+     * - `Boolean` : 1.0 (true) or 0.0 (false)
+     * - `Duration` : milliseconds value
+     * - `Double`: normalized to 0.0 if below threshold
+     *
+     * @param metricValue The metric value to process
+     * @return The processed value in EMF-compatible format
+     */
+    private Object processValue(Object metricValue) {
+        if (metricValue == null) {
+            return 0.0;
+        }
+
+        // Changes boolean value from true/false to 1/0
+        if (metricValue instanceof Boolean) {
+            return metricValue.equals(true) ? 1.0 : 0.0;
+        }
+
+        // Changes duration value to a number of ms
+        if (metricValue instanceof Duration) {
+            String durationStr = metricValue.toString();
+            metricValue = Double.parseDouble(durationStr.substring(2, durationStr.length() - 1)) * 1000;
+        }
+
+        //normalize
+        if (metricValue instanceof Double) {
+            double doubleValue = (double) metricValue;
+            if (Math.abs(doubleValue) < ZERO_THRESHOLD) {
+                metricValue = 0.0;
+            }
+        }
+
+        return metricValue;
+    }
+
+    private void writeProcessedValue(JsonWriter jsonWriter, Object processedValue) {
+        if (processedValue instanceof Double) {
+            jsonWriter.writeValue((Double) processedValue);
+        } else if (processedValue instanceof Integer) {
+            jsonWriter.writeValue((Integer) processedValue);
+        } else if (processedValue instanceof Long) {
+            jsonWriter.writeValue((Long) processedValue);
+        }
+    }
+
+
+    /**
+     * # Convert SDK Metrics to EMF Format
+     *
+     * Transforms a collection of SDK metrics into CloudWatch's Embedded Metric Format (EMF).
+     * The method processes standard SDK measurements and structures them according to
+     * CloudWatch's EMF specification.
+     *
+     * ## Example Output
+     * ```json
+     * {
+     *   "_aws": {
+     *     "Timestamp": 1672963200,
+     *     "CloudWatchMetrics": [{
+     *       "Namespace": "AwsSdk/JavaSdk2",
+     *       "Dimensions": [["ServiceId"]],
+     *       "Metrics": [{
+     *         "Name": "AvailableConcurrency",
+     *       }]
+     *     }]
+     *   },
+     *   "ServiceId": "XXXXXXXXXXXXX",
+     *   "AvailableConcurrency": 5
+     * }
+     * ```
+     *
+     * @param metricCollection Collection of SDK metrics to be converted
+     * @return List of EMF-formatted metrics ready for CloudWatch ingestion
+     */
+    public List<String> convertMetricCollectionToEmf(MetricCollection metricCollection) {
+        // Map to store aggregated metrics
+        Map<String, List<Object>> aggregatedMetrics = new HashMap<>();
```

Comment:
We should avoid using Object if possible because it's not type safe. Can we follow the same approach in cw metric publisher?